tests/integration/test_kube_ovn.py

```diff
@@ -407,6 +408,86 @@ async def test_multi_nic_ipam(multi_nic_ipam):
     assert ip_address(iface_addrs["net1"][0]["local"]) in ip_network("10.123.123.0/24")
 
 
+class TCPDumpError(Exception):
+    pass
+
+
+@retry(
+    retry=retry_if_exception_type(TCPDumpError),
+    stop=stop_after_delay(60 * 10),
+    wait=wait_fixed(1),
+    before=before_log(log, logging.INFO),
+)
+async def run_tcpdump_test(ops_test, unit, interface, capture_comparator):
+    juju_cmd = f"ssh {unit.name} -- sudo timeout 5 tcpdump -ni {interface}"
+    retcode, stdout, stderr = await ops_test.juju(
+        *shlex.split(juju_cmd),
+        check=False,
+    )
+
+    # In GH actions, the output is in stderr and stdout is empty
+    output = stdout + stderr
+    # Timeout return code is 124 when command times out
+    if retcode == 124:
+        # Last 3 lines of stdout look like this:
+        # 0 packets captured
+        # 0 packets received by filter
+        # 0 packets dropped by kernel
+        for line in output.split("\n"):
+            if "packets captured" in line:
+                captured = int(line.split(" ")[0])
+                if capture_comparator(captured):
+                    log.info(
+                        f"Comparison succeeded. Number of packets captured: {captured}"
+                    )
+                    return True
+                else:
+                    msg = f"Comparison failed. Number of packets captured: {captured}"
+                    log.info(msg)
+                    raise TCPDumpError(msg)
+
+        msg = "output did not contain the number of packets captured"
+        log.info(msg)
+        log.info(f"stdout:\n{stdout}")
+        log.info(f"stderr:\n{stderr}")
+        raise TCPDumpError(msg)
+    else:
+        msg = f"Failed to execute sudo timeout tcpdump -ni {interface} on {unit.name}"
+        log.info(msg)
+        raise TCPDumpError(msg)
+
+
+async def test_global_mirror(ops_test):
```

Comment:
you may wish to consider a change to this test

```
   ...
   assert await run_tcpdump_test(..., lambda x: x == 0)
   ...
   await kube_ovn_app.set_config(...)
   try:
      model.wait_for_idle(...)
      ...
      assert await run_tcpdump_test(..., lambda x: x > 0)
   finally:
      await kube_ovn_app.set_config(...)
      model.wait_for_idle(...)
```

I find with this approach, you are essentially always restoring the juju config in the `finally` piece of a `try` block.  

Any exceptions raised within the try will STILL get caught and bubbled out as test failures -- just now you get sanity for the next test rather than having the mirror interface still enabled

Reply:
nice, yeah that would be pesky if running locally. Will fix

Reply:
I have wrapped the related code in try/finally to prevent config from hanging around